src/spikegadgets_to_nwb/convert_ephys.py

```diff
@@ -83,6 +84,43 @@ def __init__(
         else:
             self.nwb_hw_channel_order = nwb_hw_channel_order
 
+        """split excessively large iterators into smaller ones
+        """
+        iterator_size = [neo_io._raw_memmap.shape[0] for neo_io in self.neo_io]
+        iterator_size.reverse()
+        for i, size in enumerate(
+            iterator_size
+        ):  # iterate backwards so can insert new iterators
+            if size > MAXIMUM_ITERATOR_SIZE:
+                # split into smaller iterators
+                sub_iterators = []
+                j = 0
+                previous_multiplex_state = None
+                iterator_loc = len(iterator_size) - i - 1
+                while j < size:
+                    sub_iterators.append(
+                        SpikeGadgetsRawIOPartial(
+                            self.neo_io[iterator_loc],
+                            start_index=j,
+                            stop_index=j + MAXIMUM_ITERATOR_SIZE,
+                            previous_multiplex_state=previous_multiplex_state,
+                        )
+                    )
+                    if self.n_multiplexed_channel > 0:
+                        partial_size = sub_iterators[-1]._raw_memmap.shape[0]
+                        previous_multiplex_state = sub_iterators[
+                            -1
+                        ].get_analogsignal_multiplexed_partial(
+                            i_start=partial_size - 10,
+                            i_stop=partial_size,
+                            padding=30000,
```

Comment:
Is this padding related to the sampling rate? What if there is a different sampling rate in the data?

Reply:
For example, there are some datasets with 20 kHz sampling rate